crypto/src/mls/client/key_package.rs

```diff
@@ -416,6 +417,69 @@ mod tests {
         assert!(Client::is_mls_keypackage_expired(&kp_1s_exp));
     }
 
+    #[apply(all_cred_cipher)]
+    #[wasm_bindgen_test]
+    async fn requesting_x509_key_packages_after_basic(case: TestCase) {
+        // Basic test case
+        if !case.is_basic() {
+            return;
+        }
+        run_test_with_client_ids(case.clone(), ["alice"], move |[mut client_context]| {
+            Box::pin(async move {
+                let signature_scheme = case.signature_scheme();
+                let cipher_suite = case.ciphersuite();
+
+                // Generate 5 Basic key packages first
+                let _basic_key_packages = client_context
+                    .mls_central
+                    .get_or_create_client_keypackages(cipher_suite, MlsCredentialType::Basic, 5)
+                    .await
+                    .unwrap();
+
+                // Set up E2E identity
+                let new_signature_scheme = case.signature_scheme();
```

Comment:
We call `case.signature_scheme()` here, but we already called it before and assigned the result to `signature_scheme`. Do we expect that `new_signature_scheme != signature_scheme`?